edgelet/hsm-sys/azure-iot-hsm-c/src/certificate_info.c

```diff
@@ -92,6 +91,23 @@ static const int month_day[] = { 0, 31, 59, 90, 120, 151, 181, 212, 243, 273, 30
 #define END_HEADER_LENGTH   25 // length of end header string -----END CERTIFICATE-----
 #define INVALID_TIME        -1
 
+// https://tools.ietf.org/html/rfc5280#appendix-A
+#define MAX_LEN_COMMON_NAME 64
+// Common name OID 2.5.4.3
+#define COMMON_NAME_OID_SIZE 5
+static const unsigned char COMMON_NAME_OID[COMMON_NAME_OID_SIZE] = {  0x06, 0x03, 0x55, 0x04, 0x03 };
+
+#define FREEIF(x) \
+    do { \
+        if ((x != NULL)) { \
+            free(x); \
```

Comment:
Isn't passing `NULL` for `free` allowed? From the standard:
> **7.22.3.3**
> The `free` function causes the space pointed to by `ptr` to be deallocated, that is, made available for further allocation.  If `ptr` is a null pointer, no action occurs.

Reply:
Cool good to know but I don't feel like changing this.